docker.go

```diff
@@ -1027,6 +1015,27 @@ func (p *DockerProvider) CreateContainer(ctx context.Context, req ContainerReque
 						}
 					}
 
+					return nil
+				},
+			},
+			PostStarts: []ContainerHook{
+				// first post-start hook is to wait for the container to be ready
+				func(ctx context.Context, c Container) error {
+					dockerContainer := c.(*DockerContainer)
```

Comment:
I'm thinking about adding a defensive check for this type cast 🤔 

```suggestion
					if dockerContainer, ok := c.(*DockerContainer); !ok {
						// log that custom implementations of Container are not allowed?
						return nil
					}
```

Reply:
OR we simply ignore my previous comment, because we already assume that it's a Docker container here: https://github.com/testcontainers/testcontainers-go/pull/1297/files#diff-b68159b6168f099a639e8c6f96d75f23333ab0f0505ee22a98268748db17ae4cL204